pyomo/solver/ipopt.py

```diff
@@ -0,0 +1,517 @@
+        )
+        # TODO: Add in a deprecation here for keepfiles
+        self.temp_dir: str = self.declare(
+            'temp_dir', ConfigValue(domain=str, default=None)
+        )
+        self.solver_output_logger = self.declare(
+            'solver_output_logger', ConfigValue(default=logger)
+        )
+        self.log_level = self.declare(
+            'log_level', ConfigValue(domain=NonNegativeInt, default=logging.INFO)
+        )
+
+
+class ipoptResults(Results):
+    def __init__(
+        self,
+        description=None,
+        doc=None,
+        implicit=False,
+        implicit_domain=None,
+        visibility=0,
+    ):
+        super().__init__(
+            description=description,
+            doc=doc,
+            implicit=implicit,
+            implicit_domain=implicit_domain,
+            visibility=visibility,
+        )
+        self.timing_info.no_function_solve_time: Optional[
+            float
+        ] = self.timing_info.declare(
+            'no_function_solve_time', ConfigValue(domain=NonNegativeFloat)
+        )
+        self.timing_info.function_solve_time: Optional[
+            float
+        ] = self.timing_info.declare(
+            'function_solve_time', ConfigValue(domain=NonNegativeFloat)
+        )
+
+
+class ipoptSolutionLoader(SolutionLoaderBase):
+    pass
+
+
+ipopt_command_line_options = {
+    'acceptable_compl_inf_tol',
+    'acceptable_constr_viol_tol',
+    'acceptable_dual_inf_tol',
+    'acceptable_tol',
+    'alpha_for_y',
+    'bound_frac',
+    'bound_mult_init_val',
+    'bound_push',
+    'bound_relax_factor',
+    'compl_inf_tol',
+    'constr_mult_init_max',
+    'constr_viol_tol',
+    'diverging_iterates_tol',
+    'dual_inf_tol',
+    'expect_infeasible_problem',
+    'file_print_level',
+    'halt_on_ampl_error',
+    'hessian_approximation',
+    'honor_original_bounds',
+    'linear_scaling_on_demand',
+    'linear_solver',
+    'linear_system_scaling',
+    'ma27_pivtol',
+    'ma27_pivtolmax',
+    'ma57_pivot_order',
+    'ma57_pivtol',
+    'ma57_pivtolmax',
+    'max_cpu_time',
+    'max_iter',
+    'max_refinement_steps',
+    'max_soc',
+    'maxit',
+    'min_refinement_steps',
+    'mu_init',
+    'mu_max',
+    'mu_oracle',
+    'mu_strategy',
+    'nlp_scaling_max_gradient',
+    'nlp_scaling_method',
+    'obj_scaling_factor',
+    'option_file_name',
+    'outlev',
+    'output_file',
+    'pardiso_matching_strategy',
+    'print_level',
+    'print_options_documentation',
+    'print_user_options',
+    'required_infeasibility_reduction',
+    'slack_bound_frac',
+    'slack_bound_push',
+    'tol',
+    'wantsol',
+    'warm_start_bound_push',
+    'warm_start_init_point',
+    'warm_start_mult_bound_push',
+    'watchdog_shortened_iter_trigger',
+}
+
+
+@SolverFactory.register('ipopt_v2', doc='The ipopt NLP solver (new interface)')
+class ipopt(SolverBase):
+    CONFIG = ipoptConfig()
+
+    def __init__(self, **kwds):
+        self._config = self.CONFIG(kwds)
```

Comment:
This will get moved to the base class, and we will super.__init__ the base class.